ir/src/boundary_constraints.rs

```diff
@@ -95,6 +95,40 @@ impl BoundaryConstraints {
                 },
                 _ => unimplemented!(),
             },
+            IdentifierType::TraceColumnGroup(column_group) => {
+                if let TraceColAccess::GroupAccess(name, idx) = constraint.column() {
+                    if *idx >= column_group.size() {
+                        return Err(SemanticError::vector_access_out_of_bounds(
+                            &VectorAccess::new(name.clone(), *idx),
+                            column_group.size(),
+                        ));
```

Comment:
this should probably be `trace_access_out_of_bounds` instead of vector access

Reply:
Thanks, fixed.